app/src/main/java/com/example/demomvp/data/source/repository/ChampionRepository.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.demomvp.data.source.repository
+
+import com.example.demomvp.data.model.Champion
+import com.example.demomvp.data.source.local.ChampionLocalDataSource
+import com.example.demomvp.data.source.remote.ChampionRemoteDataSource
+import com.example.demomvp.data.source.remote.OnFetchDataJsonListener
+
+class ChampionRepository private constructor(private val remote: ChampionDataSource.Remote,
+                                             private val local: ChampionDataSource.Local) {
+
+    private object Holder {
+        val INSTANCE = ChampionRepository(
+            remote = ChampionRemoteDataSource.instance,
+            local = ChampionLocalDataSource.instance
+        )
+    }
+
+    companion object {
+        val instance: ChampionRepository by lazy { Holder.INSTANCE }
+    }
+
+    fun getChampion(listener: OnFetchDataJsonListener<MutableList<Champion>>) {
+        remote.getChampions(listener)
+    }
```

Comment:
Mình suggest bạn thiết kế singleton theo cách sau khi có tham số khác truyền vào nhé
Truyền thông qua constructor thay vì khởi tạo đối tượng trong class để tránh sự phụ thuộc lẫn nhau
Theo nguyên lý D trong SOLID
class SingletonObj {
    private constructor(context: Context)

    companion object {
        @Volatile private var mInstance: SingletonObj? = null

        public  fun get(context: Context): SingletonObj =
            mInstance ?: synchronized(this) {
                val newInstance = mInstance ?: SingletonObj(context).also { mInstance = it }
                newInstance
            }
    }
}